lib/plugins/aws/package/compile/events/alb/index.test.js

```diff
@@ -79,4 +81,267 @@ describe('AwsCompileAlbEvents', () => {
     });
   });
+
+  describe('onUnauthenticatedRequest', () => {
+    let cfResources;
+    let naming;
+
+    const serverlessConfiguration = (authorizerConfigOverride = {}) => ({
+      provider: {
+        alb: {
+          authorizers: {
+            test: Object.assign(
+              {
+                type: 'cognito',
+                userPoolClientId: 'userPoolClientId',
+                userPoolArn: 'userPoolArn',
+                userPoolDomain: 'userPoolDomain',
+                scope: 'openid',
+                sessionCookieName: 'sessionCookie',
+              },
+              authorizerConfigOverride
+            ),
+          },
+        },
+      },
+      functions: {
+        trigger: {
+          events: [
+            {
+              alb: {
+                listenerArn:
+                  'arn:aws:elasticloadbalancing:' +
+                  'us-east-1:123456789012:listener/app/my-load-balancer/' +
+                  '50dc6c495c0c9188/f2f7dc8efc522ab2',
+                conditions: {
+                  path: '/',
+                },
+                priority: 1,
+                authorizer: 'test',
+              },
+            },
+          ],
+        },
+      },
+    });
+
+    const baseAuthenticateCognitoConfig = (override = {}) =>
+      Object.assign(
+        {
+          UserPoolArn: 'userPoolArn',
+          UserPoolClientId: 'userPoolClientId',
+          UserPoolDomain: 'userPoolDomain',
+          OnUnauthenticatedRequest: 'deny',
+          Scope: 'openid',
+          SessionCookieName: 'sessionCookie',
+          AuthenticationRequestExtraParams: undefined,
+          SessionTimeout: undefined,
+        },
+        override
+      );
+
+    it('maps undefined allowUnauthenticated to "deny"', () =>
+      fixtures
+        .extend('functionDestinations', serverlessConfiguration())
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          })
+        )
+        .then(serverless => {
+          ({ Resources: cfResources } = serverless.service.provider.compiledCloudFormationTemplate);
+          naming = serverless.getProvider('aws').naming;
+        })
+        .then(() => {
+          const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId('trigger', 1);
+          const rule = cfResources[albListenerRuleLogicalId];
+
+          expect(rule.Type).to.equal('AWS::ElasticLoadBalancingV2::ListenerRule');
+          expect(rule.Properties.Actions).to.have.length(2);
+          expect(rule.Properties.Actions[0].Type).to.equal('authenticate-cognito');
+          expect(rule.Properties.Actions[0].Order).to.equal(1);
+          expect(rule.Properties.Actions[0].AuthenticateCognitoConfig).to.deep.equal(
+            baseAuthenticateCognitoConfig()
+          );
+
+          expect(rule.Properties.Actions[1].Type).to.equal('forward');
+          expect(rule.Properties.Actions[1].Order).to.equal(2);
```

Comment:
It'll be nice if this test matches one that confirms for `allow`, so just checks for `deny`.

I wouldn't check for every CF property, as that technically means reimplementing CF configuration in unit tests, and it's not what I think tests should be about (Testing that given CF configuration results with desired outcome is in scope of integration tests)